### DISCUSSION TITLE
How to add Python modules to Spyder & Jupyter installed by brew

### DISCUSSION THREAD

**Discussion by cuga-rajal** [Homebrew/discussions#5718] [Getting started]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 824efa8836dc226aa92dbbf7404c1b4a66707cca
Last commit: 3 days ago
Core tap JSON: 05 Nov 21:03 UTC
Core cask tap JSON: 05 Nov 21:03 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1-arm64
CLT: 16.0.0.26.4
Xcode: 16.0
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I installed Spyder and Jupyter with brew in macOS 15.1, and both are working well. But I need to add some python modules to their environment, and my attempts to add modules using 

	/opt/homebrew/bin/pip3 install [module]

return the "error: externally-managed-environment" message.

If I use pipx, as the error message suggests, the modules are installed in ~/.local  and they don't get loaded by Spyder or Jupyter.

I would like to take the simplest approach (without problems) to install extra python modules into the environment that Jupyter and Spyder already use. Or, somehow get Spyder and Jupyter to load these modules on startup.

If the solution is to use 

	python3 -m venv path/to/venv

I don't know what to specify for the path that's used by Spyder and Jupyter. Any help appreciated.

**Comment by dcurrie** [comment]
I am also confused about this. My workaround was to force it using `--user --break-system-packages` as in:

```
pip3 install --user --break-system-packages jupyterlab_widgets
```

It will probably lead to compatibility problems later when brew updates, so it is only a workaround.

`jupyterlab` supports virtual environments, but only once it's running. I wasn't able to get `jupyterlab_widgets` from the venv; it seems `jupyter` looks into its own env for plugins.
